src/test/java/org/prebid/server/auction/PrivacyEnforcementServiceTest.java

```diff
@@ -0,0 +1,615 @@
+
+        final BidRequest bidRequest = givenBidRequest(givenSingleImp(
+                singletonMap(BIDDER_NAME, 1)),
+                bidRequestBuilder -> bidRequestBuilder
+                        .user(user)
+                        .device(device)
+                        .regs(regs));
+
+        // when
+        final Future<Map<String, PrivacyEnforcementResult>> masked = privacyEnforcementService
+                .mask(bidderToUser, null, extRegs, singletonList(BIDDER_NAME), emptyMap(), bidRequest, PUBLISHER_ID, timeout);
+
+        // then
+        final PrivacyEnforcementResult expected = PrivacyEnforcementResult.of(user, device, regs);
+
+        verify(gdprService).isGdprEnforced(eq("0"), eq(PUBLISHER_ID), eq(singleton(15)), eq(timeout));
+        verifyNoMoreInteractions(gdprService);
+
+        assertThat(masked.succeeded()).isTrue();
+        assertThat(masked.result()).hasSize(1)
+                .containsOnly(entry(BIDDER_NAME, expected));
+    }
+
+    @Test
+    public void shouldResolveBidderNameByAliases() {
+        // given
+        given(gdprService.isGdprEnforced(eq("1"), eq(PUBLISHER_ID), eq(singleton(15)), eq(timeout)))
+                .willReturn(Future.succeededFuture(true));
+        given(gdprService.resultByVendor(eq(singleton(15)), eq("1"), any(), any(), eq(timeout)))
+                .willReturn(Future.succeededFuture(GdprResponse.of(true, singletonMap(15, false), null)));
+
+        final User user = notMaskedUser();
+        final Device device = notMaskedDevice();
+        final ExtRegs extRegs = ExtRegs.of(1);
+        final Regs regs = Regs.of(null, mapper.valueToTree(extRegs));
+        final String alias = "alias";
+        final Map<String, User> bidderToUser = singletonMap(alias, user);
+        final Map<String, String> aliases = singletonMap(alias, BIDDER_NAME);
+
+        final BidRequest bidRequest = givenBidRequest(givenSingleImp(
+                singletonMap(BIDDER_NAME, 1)),
+                bidRequestBuilder -> bidRequestBuilder
+                        .user(user)
+                        .device(device)
+                        .regs(regs));
+
+        // when
+        final Future<Map<String, PrivacyEnforcementResult>> masked = privacyEnforcementService
+                .mask(bidderToUser, null, extRegs, singletonList(BIDDER_NAME), aliases, bidRequest, PUBLISHER_ID, timeout);
+
+        // then
+        final Regs expectedRegs = Regs.of(null, mapper.valueToTree(ExtRegs.of(1)));
+        final PrivacyEnforcementResult expected = PrivacyEnforcementResult.of(userGdprMasked(), deviceGdprMasked(),
+                expectedRegs);
+
+        verify(bidderCatalog, times(2)).bidderInfoByName(BIDDER_NAME);
+        verifyNoMoreInteractions(bidderCatalog);
+        verify(gdprService).isGdprEnforced(eq("1"), eq(PUBLISHER_ID), eq(singleton(15)), eq(timeout));
+        verify(gdprService).resultByVendor(eq(singleton(15)), eq("1"), any(), any(), eq(timeout));
```

Comment:
Same as for ` verify(gdprService).isGdprEnforced` verification with specific stub - it's either one or another.